Dark mode: Review detail

0nko opened the pull request:
Fixes #1885. 

I've kept the original toggle button for now.

![image](https://user-images.githubusercontent.com/1522856/73794143-1031d300-47a8-11ea-8156-ecbd5f10cf86.png)

![image](https://user-images.githubusercontent.com/1522856/73794151-1a53d180-47a8-11ea-91a7-e4f8d3f3e0c4.png)

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AmandaRiu commented on WooCommerce/src/main/res/layout/fragment_review_detail.xml:
```diff
@@ -149,38 +145,36 @@
-                android:layout_marginEnd="@dimen/margin_extra_large"
+                android:layout_marginBottom="@dimen/minor_100"
+                android:layout_marginEnd="@dimen/major_100"
                 android:background="@android:color/transparent"
```
It's so strange that there isn't a `MaterialToggleButton` (which is why some of the `Woo.Button` style settings aren't transferring properly). Recommend moving this `android:background="@android:color/transparent"` to inside the new `Woo.Button.Toggle` style for now. Hopefully someday they create a Material option.

AmandaRiu commented on WooCommerce/src/main/res/layout/fragment_review_detail.xml:
```diff
@@ -12,135 +12,131 @@
+                        android:layout_height="@dimen/image_minor_100"
                         android:importantForAccessibility="no"
                         android:scaleType="centerCrop"
                         android:src="@drawable/ic_product"/>
```
`ic_product.xml` is using `wc_border_color` (#E6E6E6) and likely needs to be updated for light/dark mode : 

![Screen Shot 2020-02-04 at 8 16 06 PM](https://user-images.githubusercontent.com/5810477/73809826-b6261300-4791-11ea-9cca-b80c047cb717.png)

@Garance91540 WDYT?

Garance91540 replied:
I'd say Status Gray Gray 5 • #DCDCDE • rgb(220,220,222) as well!

AmandaRiu commented on WooCommerce/src/main/res/values/styles_base.xml:
```diff
@@ -80,6 +80,7 @@ theme across the entire app. Overridden versions should be added to the styles.x
         Card Styles
     -->
     <style name="Woo.Card" parent="@style/Widget.MaterialComponents.CardView">
+        <item name="android:background">@color/card_bgColor</item>
```
This should be removed. This style should only be used with `MaterialCardView` which will take care of setting elevation and the appropriate alpha.

AmandaRiu commented on WooCommerce/src/main/res/values/colors.xml:
```diff
@@ -58,7 +58,7 @@
 
     <!-- Default Card colors -->
-    <color name="card_bgColor">@color/white</color>
+    <color name="card_bgColor">@color/color_surface</color>
```
Colors in the `colors.xml` file should not be edited. These are all old styles. Any new colors needed should be added to `colors_base.xml`. Either way this change is not needed - see code comments in the `styles_base.xml` file.

0nko replied:
🤦‍♂ You're right!

Garance91540 commented on WooCommerce/src/main/res/layout/fragment_review_detail.xml:
```diff
@@ -12,135 +12,131 @@
                 android:importantForAccessibility="no"
                 app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toBottomOf="@+id/divider"
                 tools:srcCompat="@drawable/ic_user_circle_grey_24dp"/>
```
I'd say Status Gray Gray 5 • #DCDCDE • rgb(220,220,222)

0nko commented:
@kyleaparker Yeah, the dark version uses a shade of orange. The light version uses the color from Figma (Reviews v1), but that would look too dark on a dark background. Do you want me to change it?

kyleaparker commented:
>@kyleaparker Yeah, the dark version uses a shade of orange. The light version uses the color from Figma (Reviews v1), but that would look too dark on a dark background. Do you want me to change it?

Yep, let's use the Muriel light/dark yellows from here: https://woomobiledesignhandbook.wordpress.com/colors/

0nko commented on WooCommerce/src/main/res/layout/fragment_review_detail.xml:
```diff
@@ -12,135 +12,131 @@
                 android:gravity="center_vertical"
                 android:orientation="horizontal"
-                android:padding="@dimen/margin_medium"
+                android:padding="@dimen/minor_100"
```
I didn't add or remove paddings/margins, I was just replacing the existing resource references with the new ones.

0nko replied:
The paddings are necessary here for the proper selectable background and margin at the top adds symmetry to the bottom margin (declared by the view below).

AmandaRiu replied:
Ah, yup, I see what you mean. 👍

0nko commented:
Thanks for the review, @AmandaRiu! I've addressed your feedback comments and the PR's ready for another round of review.

I've updated the product placeholder icon, which affects other screens, such as the product list. I've changed the icon color to On Surface / Medium Emphasis (both product icon and avatar icon) and now a different color is displayed based on the current theme:

![image](https://user-images.githubusercontent.com/1522856/73930309-99dfbe80-48d6-11ea-9415-088929122e3e.png)
![image](https://user-images.githubusercontent.com/1522856/73930364-aa903480-48d6-11ea-9f0e-8a682b2b3b6b.png)

![image](https://user-images.githubusercontent.com/1522856/73930554-f8a53800-48d6-11ea-9835-f9cc29e7e6b4.png)
![image](https://user-images.githubusercontent.com/1522856/73930578-05c22700-48d7-11ea-8897-1a262246de3f.png)

cc: @kyleaparker @Garance91540
